bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/discovery/GroupePSADiscoveryService.java

```diff
@@ -0,0 +1,94 @@
+import org.openhab.binding.groupepsa.internal.bridge.GroupePSABridgeHandler;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.Vehicle;
+import org.openhab.binding.groupepsa.internal.rest.exceptions.GroupePSACommunicationException;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link GroupePSADiscoveryService} is responsible for discovering new
+ * vehicles available for the configured app key.
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+@NonNullByDefault
+public class GroupePSADiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(GroupePSADiscoveryService.class);
+
+    private final GroupePSABridgeHandler bridgeHandler;
+
+    public GroupePSADiscoveryService(GroupePSABridgeHandler bridgeHandler) {
+        super(Collections.singleton(THING_TYPE_VEHICLE), 10, false);
+        this.bridgeHandler = bridgeHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        try {
+            List<Vehicle> vehicles = bridgeHandler.getVehicles();
+            if (vehicles == null || vehicles.isEmpty()) {
+                logger.warn("No vehicles found");
+                return;
+            }
+            for (Vehicle vehicle : vehicles) {
+                ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+                ThingTypeUID thingTypeUID = THING_TYPE_VEHICLE;
+                String id = vehicle.getId();
+                if (id != null) {
+                    ThingUID vehicleThingUid = new ThingUID(THING_TYPE_VEHICLE, bridgeUID, id);
+
+                    Map<String, Object> properties = new HashMap<>();
+                    putProperty(properties, GroupePSABindingConstants.VEHICLE_ID, id);
+                    putProperty(properties, GroupePSABindingConstants.VEHICLE_VIN, vehicle.getVin());
+                    putProperty(properties, GroupePSABindingConstants.VEHICLE_VENDOR, vehicle.getBrand());
+                    putProperty(properties, GroupePSABindingConstants.VEHICLE_MODEL, vehicle.getLabel());
+
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(vehicleThingUid)
+                            .withThingType(thingTypeUID).withProperties(properties).withBridge(bridgeUID)
+                            .withRepresentationProperty(GroupePSABindingConstants.VEHICLE_VIN)
+                            .withLabel(vehicle.getBrand() + "  (" + vehicle.getVin() + ")").build();
+
+                    thingDiscovered(discoveryResult);
+                }
+            }
+        } catch (GroupePSACommunicationException e) {
+            logger.warn("No vehicles found", e);
```

Comment:
Are you logging the stack trace by intention?

Reply:
Yes, there are various reasons why the GroupePSACommunicationException can be thrown.